src/rt/typeinfo/ti_n.d

```diff
@@ -0,0 +1,58 @@
+/**
+ * TypeInfo support code.
+ *
+ * Copyright: Copyright Digital Mars 2016.
+ * License:   $(WEB www.boost.org/LICENSE_1_0.txt, Boost License 1.0).
+ * Authors:   Kenji Hara
+ */
+
+/*          Copyright Digital Mars 2016.
+ * Distributed under the Boost Software License, Version 1.0.
+ *    (See accompanying file LICENSE or copy at
+ *          http://www.boost.org/LICENSE_1_0.txt)
+ */
+module rt.typeinfo.ti_n;
+
+// typeof(null)
+
+class TypeInfo_n : TypeInfo
+{
+    override string toString() const @safe { return "typeof(null)"; }
+
+    override size_t getHash(in void* p) const
+    {
+        return 0;
+    }
+
+    override bool equals(in void* p1, in void* p2) const @trusted
+    {
+        //return *cast(typeof(null)*)p1 is *cast(typeof(null)*)p2;
+        return true;
```

Comment:
What I'm concern is, how these TypeInfo member functions should work. As I already said, returning constant is logically valid. But if they should compare the actual runtime values, they are not good. For example:

```d
int x, y;
auto n1 = cast(typeof(null))&x;
auto n2 = cast(typeof(null))&y;
assert(typeid(typeof(null)).equals(&n1, &n2) == ?);
```